pdns/slavecommunicator.cc

```diff
@@ -1327,7 +1331,7 @@ void CommunicatorClass::slaveRefresh(PacketHandler *P)
     }
     else if(hasSOA && theirserial == ourserial) {
       uint32_t maxExpire=0, maxInception=0;
-      if(dk.isPresigned(di.zone)) {
+      if(dk.isPresigned(di.zone) && ::arg().mustDo("compare-signatures-on-zone-freshness-check")) {
```

Comment:
```suggestion
      if(::arg().mustDo("compare-signatures-on-zone-freshness-check") && dk.isPresigned(di.zone)) {
```
will save a huge amount of unnecessary db queries when the check is disabled.

Reply:
thanks, addressed